# -*- coding: utf-8 -*-
# Generated by Django 1.9.5 on 2016-07-18 07:47
from __future__ import unicode_literals

from django.conf import settings
import django.core.validators
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
        ('registrations', '0005_auto_20160708_1511'),
    ]

    operations = [
        migrations.CreateModel(
            name='CollaboratingOrganization',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('org_name', models.CharField(max_length=60, unique=True)),
            ],
        ),
        migrations.CreateModel(
            name='ExceptionListOrgApplicants',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('employee_email', models.EmailField(max_length=50, unique=True)),
                ('exception_type', models.CharField(choices=[(None, 'FEE TYPE'), ('1', 'APPLICATION FEE')], max_length=2)),
                ('employee_id', models.CharField(max_length=15)),
                ('employee_name', models.CharField(max_length=50)),
                ('org', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='exceptionlistorgapplicants_requests_created_100', to='registrations.CollaboratingOrganization')),
            ],
            options={
                'verbose_name': 'Exception List Org Applicants',
                'verbose_name_plural': 'Exception List Org Applicants',
            },
        ),
        migrations.CreateModel(
            name='Reviewer',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('reviewer', models.BooleanField(default=False)),
                ('user', models.OneToOneField(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL)),
            ],
        ),
        migrations.AddField(
            model_name='studentcandidatequalification',
            name='other_degree',
            field=models.CharField(blank=True, max_length=30, null=True),
        ),
        migrations.AddField(
            model_name='studentcandidatequalification',
            name='other_discipline',
            field=models.CharField(blank=True, max_length=40, null=True),
        ),
        migrations.AlterField(
            model_name='program',
            name='program_code',
            field=models.CharField(max_length=6, unique=True, validators=[django.core.validators.RegexValidator(code='invalid_program_code', message='program code must be Alphanumeric', regex='^[a-zA-Z0-9]+\\w*$')]),
        ),
        migrations.AlterField(
            model_name='program',
            name='program_name',
            field=models.CharField(max_length=60, unique=True),
        ),
        migrations.AddField(
            model_name='exceptionlistorgapplicants',
            name='program',
            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='exceptionlistorgapplicants_requests_created_101', to='registrations.Program'),
        ),
    ]
